diktat-api/src/main/kotlin/org/cqfn/diktat/api/DiktatRuleConfig.kt

```diff
@@ -0,0 +1,19 @@
+package org.cqfn.diktat.api
+
+import kotlinx.serialization.Serializable
+
+/**
+ * Configuration of individual [DiktatRule]
+ *
+ * @property name name of the rule
+ * @property enabled
+ * @property configuration a map of strings with configuration options
+ * @property ignoreAnnotated if a code block is marked with these annotations - it will not be checked by this rule
+ */
+@Serializable
+data class DiktatRuleConfig(
+    val name: String,
```

Comment:
In my dream I was dreaming about the decoding of ENUMs with rule names out of the box by serialization. The only problem that I see here is closest rule name suggestion algorithm provided by @petertrr . But I think this also can have workaround

Reply:
Hm, by the way - it is a good idea to suggest closest enum in ktoml out of the box 🤔